lib/model-builder.js

```diff
@@ -482,7 +477,7 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
 
       // in case no merge policy matched, apply a deep merge
       // this for example handles {replace: false} and {rank: true}
-      mergedSettings[key] = deepMerge(baseClassSettings[key], subClassSettings[key]);
+      mergedSettings[key] = deepMergeProperty(baseClassSettings[key], subClassSettings[key]);
```

Comment:
I had to introduce this new helper which is a safer version of deepMerge to work with object properties of any type.
`utils.deepMerge` actually turns a provided extra to `{}` if not an Array, which is fine when passing an object {} as extras, but is not wanted when passing a simple type as a string.